### DISCUSSION TITLE
program initializatiin

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#48] [Ideas]
I wonder if initialization of specific program could be delayed to the first used of the program to speed boot time?

**Comment by VendelinSlezak** [comment]
Initalizing of programs takes very little time, not even up to one second, because it is just bunch of memory allocations and setting variabiles, so delaying it would not really speed up booting.

**Reply by defdefred** [reply]
For now yes but imagin when a lot more program will be available ;-).

**Reply by defdefred** [reply]
And also unused program shoudn't pre allocated memory.

**Reply by VendelinSlezak** [reply]
I do not think that even when there will be many programs that this will take more time than one or two seconds, what is fine. And most of programs are allocating just few KB of memory, so it do not hurt, and they are prepared to be used immediately.

**Reply by defdefred** [reply]
Make sens.
And we can imagine that big specific program, like games with heavy graphics could have a dedicated boot environment :-)

**Reply by VendelinSlezak** [reply]
Yeah, that would be pretty different story :)
